yt_dlp/extractor/bbc.py

```diff
@@ -1172,8 +1172,10 @@ def _real_extract(self, url):
                     entries, playlist_id, playlist_title, playlist_description)
 
         initial_data = self._parse_json(self._search_regex(
-            r'window\.__INITIAL_DATA__\s*=\s*({.+?});', webpage,
+            r'window\.__INITIAL_DATA__\s*=\s*("{.+?}");', webpage,
             'preload state', default='{}'), playlist_id, fatal=False)
+        if isinstance(initial_data, str):
```

Comment:
If `initial_data` is a result of parsing a JSON string literal, then this check is redundant.